java/ql/automodel/publish.sh

```diff
@@ -1,17 +1,23 @@
 #!/bin/bash
 set -e
 
-# Add help message
+help="Usage: ./publish --[override-release] --[dry-run]
+Publish the automodel query pack.
+
+If no arguments are provided, publish the version of the codeql repo specified by the latest official release of the codeml-automodel repo.
+If the --override-release argument is provided, your current local HEAD is used (for unofficial releases or patching).
+If the --dry-run argument is provided, the release is not published (for testing purposes)."
+
+# Echo the help message
 if [ "$1" = "-h" ] || [ "$1" = "--help" ]; then
-  echo "Usage: ./publish [override-release]"
-  echo "By default we publish the version of the codeql repo specified by the latest official release defined by the codeml-automodel repo."
-  echo "Otherwise, the optional argument override-release forces your current HEAD to be published."
+  echo "$help"
   exit 0
 fi
 
-# Check that either there are 0 or 1 arguments, and if 1 argument then check that it is "override-release"
-if [ $# -gt 1 ] || [ $# -eq 1 ] && [ "$1" != "override-release" ]; then
-  echo "Error: Invalid arguments. Please run './publish --help' for usage information."
+# Check the arguments are valid
+if [ $# -gt 2 ] || ([ $# -eq 1 ] && [ "$1" != "--override-release" ] && [ "$1" != "--dry-run" ]) || ([ $# -eq 2 ] && [ "$1" != "--override-release" ] && [ "$1" != "--dry-run" ] && [ "$2" != "--override-release" ] && [ "$2" != "--dry-run" ]); then
```

Comment:
This line is huge, hard to visually parse and will get even worse, should we add more flags.

I think it'd be more legible to use the [shift](https://unix.stackexchange.com/questions/174566/what-is-the-purpose-of-using-shift-in-shell-scripts) builtin in a loop above. Pseudo-code:

```bash
OVERRIDE_RELEASE=0
DRY_RUN=0
while [ $# > 1 ]; do
  if [ $1 = "--override-release" ]; do
    OVERRIDE_RELEASE=1
  fi
  if [ $1 = "--dry-run" ]; do
    DRY_RUN=1
  fi
  shift;
done
```

Then, you could simply use the `OVERRIDE_RELEASE` and `DRY_RUN` flags here and elsewhere, rather than needing to worry about the (semantically insignificant) order in which those flags were passed in etc.